crates/re_viewer/src/ui/override_ui.rs

```diff
@@ -0,0 +1,266 @@
+        });
+}
+
+pub fn add_new_override(
+    ctx: &ViewerContext<'_>,
+    query: &LatestAtQuery,
+    store: &DataStore,
+    ui: &mut egui::Ui,
+    space_view: &SpaceViewBlueprint,
+    data_result: &DataResult,
+    active_components: &BTreeSet<ComponentName>,
+) {
+    ui.menu_button("Add new override", |ui| {
+        ui.style_mut().wrap = Some(false);
+
+        let view_systems = ctx
+            .space_view_class_registry
+            .new_visualizer_collection(*space_view.class_identifier());
+
+        // We have to have at least 1 visualizer system or we can't create an override
+        let Some(system_for_initial_value) = view_systems.systems.values().next() else {
+            return;
+        };
+
+        let mut components_per_visualizer = data_result
+            .visualizers
+            .iter()
+            .filter_map(|vis| view_systems.get_by_identifier(*vis).ok())
+            .map(|vis| vis.visualizer_query_info().queried);
+
+        // Accmulate all the components if there are multiple visualizers
+        let mut components = components_per_visualizer.next().unwrap_or_default();
+        for mut rest in components_per_visualizer {
+            components.append(&mut rest);
+        }
+
+        // Empty space views of every available types
+        for component in components.difference(active_components) {
+            // If we don't have an override_path we can't set up an initial override
+            // this shouldn't happen if the `DataResult` is valid.
+            let Some(override_path) = data_result.override_path() else {
+                continue;
+            };
+
+            // If there is no registered editor, don't let the user create an override
+            if !ctx.component_ui_registry.has_registered_editor(component) {
+                continue;
+            }
+
+            if ui.button(component.as_str()).clicked() {
+                // We are creating a new override. We need to decide what initial value to give it.
+                // - First see if there's an existing splat in the recording.
+                // - Next see if visualizer system wants to provide a value.
+                // - Finally, fall back on the default value from the component registry.
+
+                let components = [*component];
+
+                let mut splat_cell: DataCell = [InstanceKey::SPLAT].into();
+                splat_cell.compute_size_bytes();
+
+                let Some(mut initial_data) = store
+                    .latest_at(query, &data_result.entity_path, *component, &components)
+                    .and_then(|result| result.2[0].clone())
+                    .and_then(|cell| {
+                        if cell.num_instances() == 1 {
+                            Some(cell)
+                        } else {
+                            None
+                        }
+                    })
+                    .or_else(|| {
+                        system_for_initial_value.initial_override_value(
```

Comment:
this doesn't check out for me: The system for the initial value may not have brought up the component that we're on right now.
I think what you iterate instead is pairs of component-visualizer where you somehow removed duplicated components, i.e. decided which visualizer would provide the initial override value :/

Reply:
Aha! Yes, good catch. Changed to build a map of Component -> Visualizer and use the right now.